py/torch_tensorrt/dynamo/utils.py

```diff
@@ -256,48 +257,54 @@ def prepare_inputs(
     inputs: Input | torch.Tensor | Sequence[Any] | Dict[Any, Any],
     disable_memory_format_check: bool = False,
 ) -> Any:
-    if inputs is None:
-        return None
-
-    elif isinstance(inputs, Input):
-        return inputs
+    """
+    We take a nested group of torch.Tensors or scalars and convert them into torchtrt.Input's
+    """
+    # Any tensors created inside this call will be FakeTensors if it's inside a torch.compile session
+    # So, we disable fake mode temporarily.
+    with unset_fake_temporarily():
+        if inputs is None:
+            return None
 
-    elif isinstance(inputs, (torch.Tensor, int, float, bool)):
-        return Input.from_tensor(
-            torch.tensor(inputs),
-            disable_memory_format_check=disable_memory_format_check,
-        )
+        elif isinstance(inputs, Input):
+            return inputs
 
-    elif isinstance(inputs, (list, tuple)):
-        torchtrt_input_list = []
-        for input_obj in inputs:
-            torchtrt_input = prepare_inputs(
-                input_obj, disable_memory_format_check=disable_memory_format_check
+        elif isinstance(inputs, (torch.Tensor, int, float, bool)):
+            return Input.from_tensor(
+                torch.tensor(inputs),
+                disable_memory_format_check=disable_memory_format_check,
             )
-            torchtrt_input_list.append(torchtrt_input)
-
-        return (
-            torchtrt_input_list
-            if isinstance(inputs, list)
-            else tuple(torchtrt_input_list)
-        )
 
-    elif isinstance(inputs, dict):
-        torchtrt_inputs_dict: Dict[Any, Any] = dict()
+        elif isinstance(inputs, (list, tuple)):
+            torchtrt_input_list = []
+            for input_obj in inputs:
+                torchtrt_input = prepare_inputs(
+                    input_obj, disable_memory_format_check=disable_memory_format_check
+                )
+                torchtrt_input_list.append(torchtrt_input)
 
-        for key, input_obj in inputs.items():
-            torchtrt_input = prepare_inputs(
-                input_obj, disable_memory_format_check=disable_memory_format_check
+            return (
+                torchtrt_input_list
+                if isinstance(inputs, list)
+                else tuple(torchtrt_input_list)
             )
-            torchtrt_inputs_dict[key] = torchtrt_input
 
-        return torchtrt_inputs_dict
+        elif isinstance(inputs, dict):
+            torchtrt_inputs_dict: Dict[Any, Any] = dict()
```

Comment:
Should this be str: Any?

Reply:
Yes, I think so. This was not changed but the linter just restructured it. I'll modify it